### PR TITLE
Add Clang tidy config

### PR DESCRIPTION
**Pull request by hannespetur**
Adds a Clang tidy config (`.clang-tidy`) in project root. PR also includes several fixes for a few of the current tidy warnings as a start (a ton of warnings still remain). A long term goal would be to fix all of those warnings and add a workflow check (similar to "check_format" with clang-format).

### CODE REVIEW

File: include/graphtyper/typer/sample_call.hpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ class SampleCall
   std::pair<uint16_t, uint16_t> get_gt_call() const;
   uint8_t get_gq() const;
   uint8_t get_lowest_phred_not_with(uint16_t allele) const;
-  int8_t check_filter(long gq) const;
+  long check_filter(long gq) const;
```
Comment (h-2) [inline include/graphtyper/typer/sample_call.hpp:48]: Why this change?

I would actually propose replacing all instances of `long` and `int` with fixed-size counter-parts or `size_t` when something is counted. This is actually quite important for serialised data structures as `long` and `int` have different sizes on different platforms.

**Review by h-2 — APPROVED** [review]
Yes, that's a very good plan! My editor has clang-tidy built-in and was already warning about some things!

Also 👍🏻 for going through the code sections that are commented out and figuring out if they can be deleted!

File: include/graphtyper/typer/sample_call.hpp (hunk 1 above)
Comment (hannespetur) [reply include/graphtyper/typer/sample_call.hpp:48]: Previous code had a "bugprone-signed-char-misuse" warning from tidy. I don't actually see anything that could potentially be a bug here but it's just some `int8_t` is `signed char` fun.

I personally think `int`/`long` are fine defaults when exact size doesn't matter. But sure we can use fixed-size types in serialized data structures though.
